### DISCUSSION TITLE
How can we check whether Chromium supports QS-algorithms?

### DISCUSSION THREAD

**Discussion by KrishCrS** [open-quantum-safe/oqs-demos#244] [Q&A] [answered]
Can you @Raytonne please suggest how to confirm whether Chromium has been properly enabled with OQS support? I've attempted to follow the **test** instructions provided in the #210, [README-Linux.md](https://github.com/open-quantum-safe/oqs-demos/commit/0b5a94fab633b1a428641de49f11f8b9031c982b#diff-9cb5f72748c8338ff4c54d34e9e18739c895eec58f7f8e91ee3342558e513b3b) onwards line number 46, but unfortunately, neither method appears to be effective.

**Comment by Raytonne** [comment]
Hi - You're using an outdated instruction. The Linux instruction has been updated. You can just follow [the instruction](https://github.com/open-quantum-safe/oqs-demos/blob/main/chromium/README-Linux.md).

If everything runs smoothly, you should be able to access ports at https://test.openquantumsafe.org/

**Reply by KrishCrS** [reply]
Hello,
A miscommunication has occurred. I followed the latest instructions (#240) to build chromium, but, I think, there is no proper instruction for testing. However, I have downloaded `CA.crt` from https://test.openquantumsafe.org/ and imported it to `chrome://settings/security-->Authorities` in the newly built Chromium, but I am still unable to access the links provided on the https://test.openquantumsafe.org/ . Can you please suggest what the problem may be?

**Reply by Raytonne** [reply]
What error did you get? e.g., ERR_SSL_VERSION_OR_CIPHER_MISMATCH?
Which port did you try to access? This chromium only supports algorithms documented [here](https://github.com/open-quantum-safe/boringssl#supported-algorithms)
Have you downloaded the latest CA.crt? We regenerated the CA.crt last Friday.

**Reply by KrishCrS** [reply]
I used latest `CA.crt` and tried ports `6014, 6040, 6058, 6083, 6109, and 6118`.
In each case, `ERR_CONNECTION_TIMED_OUT` occurred.

**Reply by KrishCrS** [reply]
```
ubuntu22@ubuntu22:~/Music/QSChromiumBuilt/chromium/src$ ./out/Default/chrome
[10106:10106:1101/170658.862055:ERROR:policy_logger.cc(154)] :components/enterprise/browser/controller/chrome_browser_cloud_management_controller.cc(163) Cloud management controller initialization aborted as CBCM is not enabled.
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
[10182:7:1101/170659.481868:ERROR:command_buffer_proxy_impl.cc(128)] ContextResult::kTransientFailure: Failed to send GpuControl.CreateCommandBuffer.
[10182:7:1101/170659.482115:ERROR:context_provider_command_buffer.cc(155)] GpuChannelHost failed to create command buffer.
[10106:10106:1101/170819.056022:ERROR:render_widget_host_view_aura.cc(463)] Not implemented reached in virtual gfx::NativeViewAccessible content::RenderWidgetHostViewAura::GetNativeViewAccessible()
```
While I run Chromium, I got the above error. Are these signs that Chromium is not built with OQS support?

Also while trying to access port `6118`, `MEDIA_ELEMENT_ERROR` is occured.
```
[10350:1:1101/171357.443874:ERROR:html_media_element.cc(4917)] SetError: {code=4, message="MEDIA_ELEMENT_ERROR: Media load rejected by URL safety check"}
```

**Reply by baentsch** [reply, marked as answer]
>  Are these signs that Chromium is not built with OQS support?

To me this looks like general setup errors but defer to @Raytonne 's expertise with the current Chromium code base. My recommendation would be that you review the referenced code and check for possible causes, @KrishCrS . FWIW, `MEDIA_ELEMENT_ERRORS` related to TLS errors (which would be indicative of missing/wrong crypto algorithms) I've seen only in conjunction with clear error messages such as

> ERROR:ssl_client_socket_impl.cc(980)] handshake failed; returned -1, SSL error code 1, net_error -113

**Reply by Raytonne** [reply]
I agree and these look like general errors. Probably @KrishCrS can try to re-do the whole building process from beginning.

**Reply by KrishCrS** [reply]
Okay, I'll rebuild it and report back. Thanks for assistance.

**Comment by KrishCrS** [comment]
Thanks for your assistance. Finally, it is built successfully and able to connect to the test ports. Thanks again.
